### ISSUE TITLE
Dropout Study

### ISSUE THREAD

**Issue by mariahellis** [fmcnallyi3/it_anisotropy#1]
I’m a physics major involved with Dr. Frank McNally’s Machine Learning group that is under Research in Physics at Mercer University! I conducted a dropout study that consisted of 6 different models based on the baseline architecture. After adding dropout parameters, I only changed the dropout percentages, with them being .04, .15, .30, .65, .75, and .90. 
The first set of models trained for 20 epochs, and what I found was that the models with the .04 and .90 parameters (the two extremes) performed the best and were the most comparable, with the other models being significantly less accurate. 
 
![Image](https://github.com/user-attachments/assets/81550bdf-f007-4816-9c15-eeb233dd829c)

They also did not train for the full 20 epochs:
 
![Image](https://github.com/user-attachments/assets/41094d4a-15d4-4cbb-904b-3f6afbfbaa27)

![Image](https://github.com/user-attachments/assets/78622900-1085-4f3a-813b-e280933fe7f1)

 

To see if this was an element of randomness, I made 6 more models with the exact same parameters as the original six but ran them for 80 epochs. 
The result was nothing like the 20-epoch result. These 6 new models performed differently; the previous two extremes were now fully comparable / similar in value to the other models. 
 
![Image](https://github.com/user-attachments/assets/f0677f4c-1d05-44e4-a751-da2e043ee875)

Now that the models were performing better, I trained ANOTHER 6 models with the exact same parameters and epochs (80) and got different results that were similar to the 20-epoch test:
 

![Image](https://github.com/user-attachments/assets/322a1713-26c4-4eee-8957-5246d8161e9e)

This means that the first 80-epoch test had an element of randomness being used, which I couldn’t replicate on the second test. To prevent different results using the same inputs, the random seeds for Python, NumPy, TensorFlow should be set early enough to prevent randomness occurring.
